simpleviewinc/keg-hub
# Checkbox fixes

## Context

* Daniel found a warning logged in one of my other PRs: https://github.com/simpleviewinc/tap-events-force/pull/112
* React requires that the `onChange` prop be set if you use the `checked` prop for the `input` checkbox, but certain conditions with our `checkbox` component caused the press handler to be undefined
* there was also a reference error in `option.js` preventing keg-components from building for me, so I fixed that

## Goal
* ensure the press handler is always defined (using `noOp`)
* fix the reference error in `option.js` (defStyles)

## Updates
* `src/components/form/checkbox/checkbox.wrapper.js`
  * updated `pressHandlerProp` to always be defined, even if it's a `noOp`
* `src/theme/form/option.js`
  * re-defined the `defStyles` variable that was removed in the previous commit

## Testing
* `keg components pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:checkbox-fixes`
* ensure all the checkbox stories behave as expected
* `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-511-session-unbook`
* follow Daniel's steps to reproduce the warning for the tap: https://github.com/simpleviewinc/tap-events-force/pull/112#discussion_r555347740
  * verify you don't see the warning
